odl/solvers/util/callback.py

```diff
@@ -325,12 +328,18 @@ def __init__(self, *args, **kwargs):
         self.kwargs = kwargs
         self.fig = kwargs.pop('fig', None)
         self.display_step = kwargs.pop('display_step', 1)
+        self.saveto = kwargs.pop('saveto', None)
         self.iter = 0
 
     def __call__(self, x):
         """Show the current iterate."""
         if (self.iter % self.display_step) == 0:
-            self.fig = x.show(*self.args, fig=self.fig, **self.kwargs)
+            if self.saveto is None:
+                self.fig = x.show(*self.args, fig=self.fig, **self.kwargs)
+            else:
+                self.fig = x.show(*self.args, fig=self.fig,
+                                  saveto=(self.saveto + '_' + str(self.iter)),
```

Comment:
On most OSs, `iter_10` comes before `iter_2`, frankly I dont know any where it doesn't. I also dont know a proper solution to this except the dicom approach:

`iter_0000010`
`iter_0000002`